urlchecker.sh

```diff
@@ -0,0 +1,57 @@
+#!/bin/bash
+# url testing
+# source urlchecker.sh
+main_dir="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
+excluded_directory="*/node_modules/*"
+all_urls=()
+all_files=()
+match_file()
```

Comment:
I'd maybe name this something different - 

also it's optional to add `function` as a prefix, but I find it useful to distinguish from normal variables

```suggestion
function print_files_for_url() {
```